iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -469,43 +492,27 @@ - (void)setStarted:(BOOL)started
   }
 }
 
-- (BOOL)start
+- (void)requestPermissionIfNeededFor:(CLLocationManager *)manager
 {
-  MWMVoidBlock doStart = ^{
-    LOG(LINFO, ("startUpdatingLocation"));
-
-    CLLocationManager * locationManager = self.locationManager;
-    if ([locationManager respondsToSelector:@selector(requestWhenInUseAuthorization)])
-      [locationManager requestWhenInUseAuthorization];
-
-    [locationManager startUpdatingLocation];
-
-    setPermissionRequested();
+  [manager requestWhenInUseAuthorization];
+  setPermissionRequested();
```

Comment:
Can you please double-check if this strange call is still necessary? Can we safely remove it?

Reply:
I removed this check.And fix another related issue:
When user install the app and on first opening see `allow to use location` system alert, he can press lock button, than open app and there will be no more alert and no permission granted (code totally skips this case):
```swift
+ (void)applicationDidBecomeActive
{
  if (isPermissionRequested() || ![FirstSession isFirstSession])
    [self start];
    [[self manager] updateFrameworkInfo];
  }
}
```
see gif:
![Simulator Screen Recording - iPhone 11 Pro - 2024-01-23 at 22 13 03](https://github.com/organicmaps/organicmaps/assets/79797627/d0ae92c0-1ba2-484f-9b0b-10d9c1d0f95d)

So I;m no even remove `setPermissionRequested();` but I also fixed it with adding `requestWhenInUseAuthorization` in `applicationDidBecomeActive`:
```swift
+ (void)applicationDidBecomeActive
{
  MWMLocationManager *manager = [self manager];
  [manager requestPermissionIfNeededFor:manager.locationManager];
  
  if (![FirstSession isFirstSession])
  {
    [self start];
    [manager updateFrameworkInfo];
  }
}
```

and the user cannot use the app without proper setting permission (it will appear until the user selects case. Second call of this method do nothing):
![Simulator Screen Recording - iPhone 11 Pro - 2024-01-23 at 22 18 53](https://github.com/organicmaps/organicmaps/assets/79797627/8bac24d9-8fae-4214-976d-449364383b17)

Reply:
Good catch!

Looks like the radar button is rotating even when permission has not been granted yet. Is it a correct behavior?